### ISSUE TITLE
Builder does not accept null for eclipse @Nullable fields

### ISSUE THREAD

**Issue by sylvainlaurent** [immutables/immutables#484]
The doc http://immutables.github.io/immutable.html#nullable-attributes says

> Nullable attributes are not required to be set using a builder, and null values are permitted to initialize them

But if I try the given example
```
import org.eclipse.jdt.annotation.Nullable;
import org.immutables.value.Value;

@Value.Immutable
interface NullAccepted {
  @Nullable Integer i1();
  @Nullable Long l2();
}
```
then the builder that is generated by versions 2.3.7 and 2.3.8 contains this `i1` method:
```
public final Builder i1(java.lang.@Nullable Integer i1) {
      this.i1 = Objects.requireNonNull(i1, "i1");
      initBits &= ~INIT_BIT_I1;
      return this;
    }
```

Notice the call to `Objects.requireNonNull` which prevents passing `null`to the `i1` method despite what the doc says.

**Comment by elucash** [comment]
Definitely, should be fixed, thank you for reporting this!

**Comment by sylvainlaurent** [comment]
apparently the problem occurs when I use `org.eclipse.jdt.annotation.Nullable`
It works as intended with `javax.annotation.Nullable`
but it's still a bug since the doc says
> Any annotation with simple name Nullable will work

**Comment by elucash** [comment]
yes, this has to work as stated, so it's a bug

**Comment by elucash** [comment]
Apparently, we lacked support for `TYPE_USE` nullable annotation such as JDT's one. Fixed in master and will be released in 2.3.9
I'll close the issue once released, thank you!

**Comment by elucash** [comment]
released in 2.3.9

**Comment by sylvainlaurent** [comment]
fix OK, thanks !
